tftest.py

```diff
@@ -305,46 +295,87 @@ def setup(self, extra_files=None, plugin_dir=None, init_vars=None,
       else:
         _LOGGER.warning('no such file {}'.format(link_src))
     self._finalizer = weakref.finalize(
-        self, self._cleanup, self.tfdir, filenames, deep=cleanup_on_exit)
-    return self.init(plugin_dir=plugin_dir, init_vars=init_vars, backend=backend)
+        self, self._cleanup, self.tfdir, filenames, self.binary, deep=cleanup_on_exit)
+    return self.init(**kw)
 
-  def init(self, input=False, color=False, force_copy=False, plugin_dir=None,
-           init_vars=None, backend=True):
+  def init(self,**kw):
     """Run Terraform init command."""
-    cmd_args = parse_args(input=input, color=color, backend=backend,
-                          force_copy=force_copy, plugin_dir=plugin_dir,
-                          init_vars=init_vars)
+    cmd_args = parse_args(**kw)
     return self.execute_command('init', *cmd_args).out
 
-  def plan(self, input=False, color=False, refresh=True, tf_vars=None, targets=None, output=False, tf_var_file=None):
+  def plan(self, output=False, **kw):
     "Run Terraform plan command, optionally returning parsed plan output."
-    cmd_args = parse_args(input=input, color=color,
-                          refresh=refresh, tf_vars=tf_vars,
-                          targets=targets,  tf_var_file=tf_var_file)
+    
+    cmd_args = parse_args(**kw)
     if not output:
       return self.execute_command('plan', *cmd_args).out
-    with tempfile.NamedTemporaryFile() as fp:
-      cmd_args.append('-out={}'.format(fp.name))
+    
+    if 'out' not in kw:
+      with tempfile.NamedTemporaryFile() as fp:
+        cmd_args.append('-out={}'.format(fp.name))
+        self.execute_command('plan', *cmd_args)
+        result = self.execute_command('show', '-no-color', '-json', fp.name)    
+    else:
```

Comment:
Nit: can we use a local variable for "kw['out'] or fp.name" and skip the `else` clause?